clientv3/client.go

```diff
@@ -235,9 +237,14 @@ func dialEndpointList(c *Client) (*grpc.ClientConn, error) {
 	return nil, err
 }
 
-// isHalted returns true if the given error and context indicate no forward
-// progress can be made, even after reconnecting.
-func isHalted(ctx context.Context, err error) bool {
-	isRPCError := strings.HasPrefix(grpc.ErrorDesc(err), "etcdserver: ")
-	return isRPCError || ctx.Err() != nil
+// haltErr returns matching rpctypes.Error, or return context errors.
+// Returned error means no forward progress can be made, even after reconnecting.
+func haltErr(ctx context.Context, err error) error {
```

Comment:
can we still use isHaltedError rather than haltErr? This function should not eat the error silently.